app/api/server/v1/email-inbox.js

```diff
@@ -19,21 +19,31 @@ API.v1.addRoute('email-inbox.list', { authRequired: true }, {
 	},
 });
 
+
 API.v1.addRoute('email-inbox', { authRequired: true }, {
 	get() {
 		try {
 			check(this.queryParams, {
 				_id: Match.Maybe(String),
+				email: Match.Maybe(String),
 			});
-			const { _id } = this.queryParams;
+			const { _id, email } = this.queryParams;
 
-			const emailInboxes = Promise.await(findOneEmailInbox({ userId: this.userId, _id }));
+			if (_id && !email) {
+				const emailInboxes = Promise.await(findOneEmailInbox({ userId: this.userId, _id }));
+				if (!emailInboxes) {
+					throw new Meteor.Error('email-inbox-not-found');
+				}
+				return API.v1.success(emailInboxes);
+			}
 
-			if (!emailInboxes) {
-				throw new Meteor.Error('email-inbox-not-found');
+			if (_id && email) {
+				const emailInboxes = Promise.await(findOneDifferentEmailInboxByEmail({ userId: this.userId, _id, email }));
+				return API.v1.success({ emailInboxes });
 			}
 
-			return API.v1.success(emailInboxes);
+			const emailInboxes = Promise.await(findOneEmailInboxByEmail({ userId: this.userId, email }));
```

Comment:
This new implementation is not necessary here. A new search endpoint will return records based on the term parameter.
You can deal with the result on the client side.